torch/csrc/jit/codegen/cuda/executor_utils.h

```diff
@@ -78,6 +70,186 @@ NvrtcFunction nvrtcCompile(
     int id,
     c10::optional<int> opt_block_size = c10::nullopt);
 
+namespace caching {
+// TODO: Largely duplicated from heuristic
+//  caching look up and the purpose is also
+//  simliar. Could consider putting some of
+//  the logic in the common space and re-use
+
+enum class CompileTimeEntryType {
+  PARALLEL_BINDING_ITERDOMAINS,
+  PARALLEL_ITER_EXTENT_MAP,
+  WARP_PADDED_PARALLEL_EXTENTS,
+  VECTORIZED_TENSOR_VALIDATION,
+  INPUT_ALIAS_INDICES,
+  OUTPUT_ALIAS_INDICES
+};
+
+class ParallelBindingIterDomains {
+ public:
+  using DataType = std::vector<IterDomain*>;
+  static const CompileTimeEntryType EntryType =
+      CompileTimeEntryType::PARALLEL_BINDING_ITERDOMAINS;
+};
+
+class ParallelIterExtentMap {
+ public:
+  using DataType =
+      std::unordered_map<ParallelType, std::vector<const kir::Val*>, TypeHash>;
+  static const CompileTimeEntryType EntryType =
+      CompileTimeEntryType::PARALLEL_ITER_EXTENT_MAP;
+};
+
+struct WarpPaddedExtentsInfo {
+  std::unordered_set<const kir::Val*> warp_padded_extent_set;
+  std::unordered_map<const kir::Val*, int64_t> warp_padded_constant;
+};
+
+class WarpPaddedParallelExtents {
+ public:
+  using DataType = WarpPaddedExtentsInfo;
+  static const CompileTimeEntryType EntryType =
+      CompileTimeEntryType::WARP_PADDED_PARALLEL_EXTENTS;
+};
+
+struct VectorizedTensorInfo {
+  std::unordered_set<TensorView*> global_inp_misaligned_tv;
+  std::unordered_set<TensorView*> global_out_misaligned_tv;
+  std::unordered_map<TensorView*, int> tv_to_vector_word_size;
+  std::vector<int> inp_misaligned_tensors_pos;
+  std::vector<int> out_misaligned_tensors_pos;
+  std::unordered_map<int, int> inp_pos_to_word_size_map_to_verify;
+  std::unordered_map<int, int> out_pos_to_word_size_map_to_verify;
+};
+
+class VectorizedTensorValidation {
+ public:
+  using DataType = VectorizedTensorInfo;
+  static const CompileTimeEntryType EntryType =
+      CompileTimeEntryType::VECTORIZED_TENSOR_VALIDATION;
+};
+
+class InputAliasIndices {
+ public:
+  using DataType = std::vector<std::pair<int, int>>;
+  static const CompileTimeEntryType EntryType =
+      CompileTimeEntryType::INPUT_ALIAS_INDICES;
+};
+
+class OutputAliasIndices {
+ public:
+  using DataType = std::unordered_set<int>;
+  static const CompileTimeEntryType EntryType =
+      CompileTimeEntryType::OUTPUT_ALIAS_INDICES;
+};
+
+//! Base abstract class for unified storage in `HeuristicSummary`,
```

Comment:
Copied from `HeuristicSummary`-related class?

Reply:
Yes the base classes are essentially the same with very minor differences. I considered sharing the implementation but the two use cases are quite distantly related and the actual shared logic isn't that much. 

Thanks for the review 👍  I will try to go over the comment and naming more before merging.